benchmark/selectors/selector.js

```diff
@@ -0,0 +1,148 @@
+"use strict";
+
+const fs = require("node:fs");
+const path = require("node:path");
+const suite = require("../document-suite");
+const { JSDOM } = require("../..");
+
+exports.querySelectorAll = function () {
+  let document, selectors, total;
+  let count = 0;
+  let cycle = 0;
+
+  return suite({
+    setup() {
+      const htmlFile = path.resolve(__dirname, "sizzle-speed/data/selector.html");
+      const domstr = fs.readFileSync(htmlFile, {
+        encoding: "utf8",
+        flag: "r"
+      });
+      const { window } = new JSDOM(domstr);
+      document = window.document;
+      const cssFile = path.resolve(__dirname, "sizzle-speed/selectors.large.css");
+      const css = fs.readFileSync(cssFile, {
+        encoding: "utf8",
+        flag: "r"
+      });
+      selectors = css.split("\n");
+      total = selectors.length;
+    },
+    fn() {
+      for (const selector of selectors) {
+        try {
+          document.querySelectorAll(selector);
+        } catch (e) {
+          count++;
+        }
+      }
+    },
+    teardown() {
+      cycle++;
+    },
+    onComplete() {
+      // eslint-disable-next-line no-console
+      console.log(`${(count / cycle).toFixed(0)}/${total} fails.`);
+    }
+  });
+};
+
+// This is only for comparison. Can remove later.
+/* eslint-disable quotes, max-len */
```

Comment:
Maybe don't disable `quotes`, and just use `npm run lint -- --fix` to fixup the single quotes?

Reply:
Done.